runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/io/source/FlinkSourceSplitEnumerator.java

```diff
@@ -111,10 +131,19 @@ public void handleSplitRequest(int subtaskId, @Nullable String requesterHostname
 
   @Override
   public void addSplitsBack(List<FlinkSourceSplit<T>> splits, int subtaskId) {
-    LOG.info("Adding splits {} back from subtask {}", splits, subtaskId);
-    List<FlinkSourceSplit<T>> splitsForSubtask =
-        pendingSplits.computeIfAbsent(subtaskId, ignored -> new ArrayList<>());
-    splitsForSubtask.addAll(splits);
+    // reshuffle splits, needed after rescaling
```

Comment:
👍 would it make sense to provide additional reasoning?

Reply:
This change could be reverted. I kept it, because it might make a difference under one improbable situation - reader returns splits to enumerator, and just after that a rescale happens. In all other cases this is equivalent to the previous implementation.

Reply:
Can you please revert it then?

Reply:
I could, but why? It helps under a rare circumstances and does no harm otherwise, so ... why revert?

Reply:
That's where I was headed. If it fixes a corner case it CAN'T be reverted and we should provide proper reasoning about the corner case to provide context for whoever touches the code next.

Reply:
I see. The code was originally to solve the rescaling issue, but the whole process works a little differently and the corner case that this might solve is highly speculative, so I reverted the code to the previous version.